### DISCUSSION TITLE
如何手动打包部署？

### DISCUSSION THREAD

**Discussion by xaseven** [lobehub/lobehub#1453] [Development Question] [answered]
### 💻 系统环境

macOS

### 📦 部署环境

Other

### 🌐 浏览器

Chrome

### 🐛 问题描述

看完整个文档也没有找到如何打包的，就是我不使用docker或者其他部署方式，在clone主分支后的代码里添加了自己的代码，然后打包发布怎么操作？pnpm run build, npx next build，都没有发现打包后的目录dist

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment, marked as answer]
next 打包后的目录在 `.next` ，剩下的自己看 nextjs 文档吧。

**Reply by ahuai** [reply]
求一份直接在服务器启动的详细教程啊  目前本地二开 但是我此前都是写java py等后端的  对前端太陌生了  现在使用docker打镜像包的方式打包  但是实在是太太太慢了  严重影响进度。

**Reply by arvinxx** [reply]
你自己研究下 docker 镜像的构建流程应该也能明白的。多用用 ai

**Comment by cy948** [comment]
不使用Docker，直接在服务器上启动有两个办法：
1. 使用`pnpm build && pnpm start` ，构建后直接启动。
2. 若需`node server.js`这样启动，可以参考`Dockerfile`中的[步骤](https://github.com/lobehub/lobe-chat/blob/04a6ef75fe9983abc8aed3910f9714dff1727a76/Dockerfile#L46C3-L46C64)。
